File: js-archive/longest-palindrome.js
var longestPalindrome = function (s) {
  let mem = {}
  s.split('').forEach(el => {
    if (mem[el] === undefined) {
      mem[el] = 1
    } else {
      mem[el] += 1
    }
  })
  let ans = 0
  let max = 0
  Object.keys(mem).forEach(k => {
    if (mem[k] % 2 === 0) {
      ans += mem[k]
    } else {
      max = 1
      ans += mem[k] - 1
    }
  })
  return (ans + max)
};

console.log(longestPalindrome("abccccdd"))

File: longest-palindrome.go
func longestPalindrome(s string) int {
    mem:=map[byte]int{}

    for i:=0; i < len(s); i++ {
        if _, isHas := mem[s[i]]; isHas {
            mem[s[i]]++
        } else {
            mem[s[i]] = 1
        }
    }

    max:=0; ans:=0

    for _, v := range mem {
        if (v % 2 == 0) {ans+=v}
        if (v % 2 == 1 && v > max) {
            max = 1
            ans+=v-1
        }
    }

    return ans + max
}